### PR TITLE
Add CRaC feature

### LINKED ISSUE #1370: Add CRaC Feature
Add a CRaC Feature

See: 

https://github.com/micronaut-projects/micronaut-crac-tests/blob/master/buildSrc/src/main/java/io/micronaut/crac/feature/CracFeature.java

https://github.com/micronaut-projects/micronaut-crac-tests/blob/master/buildSrc/src/main/java/io/micronaut/crac/feature/CracFeatureValidator.java

### PR DESCRIPTION
**Pull request by timyates**
The plugin will need releasing (and the POM update needs to happen) before this will generate valid builds

Closes #1370

### CODE REVIEW

File: starter-core/src/main/java/io/micronaut/starter/feature/crac/Crac.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    @Override
+    public boolean supports(ApplicationType applicationType) {
+        return applicationType == ApplicationType.DEFAULT || applicationType == ApplicationType.CLI;
+    }
+
+    @Override
+    public String getCategory() {
+        return Category.OTHER;
```
Comment (sdelamo) [inline starter-core/src/main/java/io/micronaut/starter/feature/crac/Crac.java]: I don't think Other is a valid category

**Review by sdelamo — APPROVED** [review]

File: starter-core/src/main/java/io/micronaut/starter/feature/crac/Crac.java (hunk 1 above)
Comment (timyates) [reply starter-core/src/main/java/io/micronaut/starter/feature/crac/Crac.java]: https://github.com/micronaut-projects/micronaut-starter/blob/a5f4cbd0420ee9774dfd12e8a36aad61f0b462d9/starter-core/src/main/java/io/micronaut/starter/feature/Category.java#L41

Or do you mean it's not the correct category?  🤔  I couldn't find one that matched up...  PACKAGING?

Comment (sdelamo) [reply starter-core/src/main/java/io/micronaut/starter/feature/crac/Crac.java]: Yes, same as the Graal feature: 

https://github.com/micronaut-projects/micronaut-starter/blob/3.6.x/starter-core/src/main/java/io/micronaut/starter/feature/graalvm/GraalVM.java#L55

Comment (sdelamo) [reply starter-core/src/main/java/io/micronaut/starter/feature/crac/Crac.java]: > https://github.com/micronaut-projects/micronaut-starter/blob/a5f4cbd0420ee9774dfd12e8a36aad61f0b462d9/starter-core/src/main/java/io/micronaut/starter/feature/Category.java#L41
> 
> Or do you mean it's not the correct category? 🤔 I couldn't find one that matched up... PACKAGING?

I mean we have a test to ensure you don't use OTHER

https://github.com/micronaut-projects/micronaut-starter/blob/3.6.x/starter-core/src/test/groovy/io/micronaut/starter/feature/CategorySpec.groovy#L11

**Pull request merged**